File: inflearn-kotlin-basic/src/main/kotlin/com/haeny/lec07/java/Lec07Main.java
package com.haeny.lec07.java;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

public class Lec07Main {

    public static void main(String[] args) throws IOException {
        JavaFilePrinter printer = new JavaFilePrinter();
        printer.readFile();
    }

    private Integer parseIntOrThrowV2(@NotNull String str) {
        try {
            return Integer.parseInt(str);
        } catch (NumberFormatException e) {
            return null;
        }
    }

    private int parseIntOrThrow(@NotNull String str) {
        try {
            return Integer.parseInt(str);
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException(String.format("주어진 %s는 숫자가 아닙니다.", str));
        }
    }
}

File: inflearn-kotlin-basic/src/main/kotlin/com/haeny/lec07/kotlin/Lec07Main.kt
package com.haeny.lec07.kotlin

import java.lang.IllegalArgumentException
import kotlin.NumberFormatException

fun main() {

}

fun parseIntOrThrowV2(str: String): Int? {
    return try {
        str.toInt()
    } catch (e: NumberFormatException) {
        null
    }
}

fun parseIntOrThrow(str: String): Int {
    try {
        return str.toInt()
    } catch (e: NumberFormatException) {
        throw IllegalArgumentException("주어진 $str 는 숫자가 아닙니다.")
    }
}